src/wp-includes/media.php

```diff
@@ -5456,11 +5460,34 @@ function wp_get_loading_attr_default( $context ) {
 		return false;
 	}
 
+	// Conditionally skip lazy-loading on images before the loop.
+	static $special_content_contexts = array(
+		'wp_get_attachment_image' => true,
+		'the_post_thumbnail'      => true,
+	);
+	if (
+		isset( $special_content_contexts[ $context ] )
+		// Only apply for main query but before the loop.
+		&& $wp_query->before_loop && is_main_query()
```

Comment:
I think @spacedmonkey's point on using the `WP_Query` method instead of the function makes sense regardless of the rest of the conversation in that comment thread.
```suggestion
		&& $wp_query->before_loop && $wp_query->is_main_query()
```